apps/epl/src/epl_traffic.erl

```diff
@@ -94,7 +90,71 @@ code_change(_OldVsn, State, _Extra) ->
+verify_subscribe_reply(Reply) ->
+    lists:all(fun(X) -> X =:= ok end, Reply).
+
+verify_if_node_is_default(Node) ->
+    verify_if_node_is_default(Node, epl:get_default_node()).
+
+verify_if_node_is_default(Node, DefaultNode)
+  when Node =:= DefaultNode ->
+    {true, Node};
+verify_if_node_is_default(Node, _DefaultNode) ->
+    {false, Node}.
+
+handle_incoming_node_info({false, _Node}, _Proplist, OldTraffic, _OldMsgPass,
+                          OldViz = undefined, _Subs) ->
+    {OldTraffic, OldViz};
+handle_incoming_node_info({true, Node}, Proplist, OldTraffic,
+                          OldMsgPass, OldViz, Subs) ->
+    {Viz1, NewTraffic} = update_traffic_graph(Node, OldTraffic,
+                                              epl_viz_map:new(Node)),
+    Viz2 = merge_existing_focused_nodes_and_conns(Viz1, OldViz),
+    %% We're starting from observed node which is our graph entry point
+    Viz3 = get_message_passing_counters(Node, Proplist, Viz2, OldMsgPass),
+    %% push an update to all subscribed WebSockets
+    JSON = epl_json:encode(Viz3, <<"traffic-info">>),
+
+    [Pid ! {data, JSON} || Pid <- Subs],
+    {NewTraffic, Viz3};
+handle_incoming_node_info({false, Node}, Proplist, OldTraffic,
+                          OldMsgPass, OldViz, _Subs) ->
+    %% We're starting from observed node which is our graph entry point
+    Viz1 = get_message_passing_counters(Node, Proplist, OldViz, OldMsgPass),
+    {OldTraffic, Viz1}.
+
+merge_existing_focused_nodes_and_conns(Viz, undefined) ->
+    Viz;
+merge_existing_focused_nodes_and_conns(Viz = #{nodes := Nodes}, OldViz) ->
+    lists:foldl(fun(Node, NewViz) ->
+                        maybe_merge_focused_nodes_and_conns(Node, NewViz, OldViz)
+                end, Viz, Nodes).
+maybe_merge_focused_nodes_and_conns(#{name := Node}, Viz, OldViz) ->
+    {Region, _} = epl_viz_map:pull_region(Node, OldViz),
+    merge_focused_nodes_and_conns(Node, Viz, Region).
+
+merge_focused_nodes_and_conns(_Node, Viz, []) ->
+    Viz;
+merge_focused_nodes_and_conns(Node, Viz, #{nodes := OldFocusedNodes,
+                                           connections := OldFocusedConns}) ->
+    {Region, Viz2 = #{nodes := Regions}} = epl_viz_map:pull_region(Node, Viz),
+    RegionNodes = finally_merge_focused_nodes(Region, OldFocusedNodes),
+    RegionNodesAndConns = finally_merge_focused_conns(RegionNodes,
+                                                      OldFocusedConns),
+    finally_merge_focused_nodes_and_conns(Viz2, RegionNodesAndConns, Regions).
+
+finally_merge_focused_nodes(Region, OldFocusedNodes) ->
+    maps:merge(Region, #{nodes => OldFocusedNodes}).
+
+finally_merge_focused_conns(Region, OldFocusedConns) ->
+    maps:merge(Region, #{connections => OldFocusedConns}).
+
+finally_merge_focused_nodes_and_conns(Viz, UpdatedRegion, Regions) ->
+    maps:merge(Viz, #{nodes => [UpdatedRegion | Regions]}).
+
 get_message_passing_counters(Node, Proplist, Vizceral, OldMsgPass) ->
```

Comment:
I think that `update_message_passing_counters` would be more proper here after your changes :)